src/devnet-utils.ts

```diff
@@ -56,94 +57,125 @@ export class DevnetUtils implements Devnet {
         return `${this.hre.starknet.networkConfig.url}`;
     }
 
-    private async withErrorHandler<T>(asyncFn: () => Promise<T>, errorMessage: string) {
+    private async requestHandler(
+        url: string,
+        method: Method,
+        errorMessage?: string,
+        data?: unknown,
+        headers?: any
```

Comment:
Where is `headers` used? If used anywhere, could it be a type other than `any`?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/274/commits/cad87ed19051704fb1e0dc240df16f8583ab878e